vassal-app/src/main/java/VASSAL/i18n/PieceI18nData.java

```diff
@@ -46,6 +46,22 @@ public void add(String value, String description) {
       properties.add(new Property(value, description));
     }
   }
+
+  public String translate(String value) {
+    String localisedValue = value;
+    String i18nKey = null;
+
+    for (final Property p : getProperties()) {
+      if (p.getName().equals(value)) {
+        i18nKey = TranslatablePiece.PREFIX + p.getName();
+      }
```

Comment:
Would `i18nKey = TranslatablePiece.PREFIX + value` be more efficient?
Also, wouldn't it make sense to exit the loop once a match has been found, or is important to use the last match if there are multiples?

Reply:
@BrentEaston You'd be the one to comment on this one, I think it came from your PR-into-PR (plus I'd recognize that Commonwealth "localised" anywhere, haha!)

Reply:
value/p.getName(): In general, that sort of thing is handled by the Java optomizer, you don't really have to worry about efficiency at that level.
Ending loop early: Theoretically, yes, but in practice no Trait has more than one translatable element, so it is moot.

Reply:
Sounds good, thanks for the explanations.

Reply:
I would put a break in that if block---otherwise that will confuse the next person to look at it.

Reply:
I have done so (and rebased the PR since it has developed conflicts with master)